src/cascadia/Remoting/Monarch.cpp

```diff
@@ -147,7 +191,18 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
     // - <none>
     void Monarch::SignalClose(const uint64_t peasantId)
     {
-        _peasants.erase(peasantId);
+        // If we are quitting we don't care about maintaining our list of
+        // peasants anymore, and don't need to notify the host that something
+        // changed.
+        if (_quitting.load(std::memory_order_acquire))

```

Comment:
I don't think it is _necessary_, but normally this would end up calling _GetNumberOfPeasants through a couple of event hops and I didn't want all of the windows to have to wait until after the `_handleQuitAll` finishes.